crates/flox-rust-sdk/src/models/publish.rs

```diff
@@ -227,8 +227,33 @@ impl<'flox> Publish<'flox, NixAnalysis> {
     }
 
     /// Copy the outputs and dependencies of the package to binary store
-    pub async fn upload_binary(self) -> Result<Publish<'flox, NixAnalysis>, PublishError> {
-        todo!()
+    pub async fn upload_binary(
+        self,
+        substituter: Option<SubstituterUrl>,
+    ) -> Result<(), PublishError> {
+        let nix: NixCommandLine = self.flox.nix(Default::default());
+        let store_paths = self.store_paths()?;
+        let copy_command = runix::command::NixCopy {
```

Comment:
Just for my understanding, it doesn't look like you're using the `--from` or `--to` arguments to `nix copy`, so how is the source/destination being handled with arguments to `nix`?

Reply:
--to was fixed in #197 i already ran into that mistakenly using --store here

Reply:
maybe we can cherry pick that commit here?

https://github.com/flox/flox/pull/197/commits/5b2e867455521db45c3b651495345262e096590f

Reply:
I'm confused as to the ordering here. What is the relationship between this PR and #197?

Reply:
197 was created by more or less

```
git checkout main
git checkout -b 197
git merge 176
git merge 196
...
other commits to get the cli working
```

I want to get in 176 and 196 into main first and then rebase 197 onto it but neither of the former had approval.